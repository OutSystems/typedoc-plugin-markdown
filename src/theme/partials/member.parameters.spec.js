describe(`[partial] member.parameters`, () => {
  test('should compile', () => {
    const context = {
      parameters: [
        {
          name: 'paramZ',
          flags: [],
          id: 350,
          parent: null,
          originalName: 'paramZ',
          kind: 32768,
          type: { type: 'intrinsic', name: 'string' },
          comment: { shortText: '', text: 'This is a string parameter.' },
          kindString: 'Parameter',
        },
        {
          name: 'paramG',
          flags: [],
          id: 351,
          parent: null,
          originalName: 'paramG',
          kind: 32768,
          type: { type: 'intrinsic', name: 'any' },
          comment: {
            shortText: '',
            text:
              'This is a parameter flagged with any.\n\n    This sentence is placed in the next line.\n',
          },
          kindString: 'Parameter',
        },
        {
          name: 'paramA',
          flags: [],
          id: 352,
          parent: null,
          originalName: 'paramA',
          kind: 32768,
          type: {
            type: 'reference',
            name: 'INameInterface',
            symbolID: -1037,
            reflection: null,
          },
          comment: {
            shortText: '',
            text:
              "\nThis is a **parameter** pointing to an interface.\n\n```\nvar value:BaseClass = new BaseClass('test');\nfunctionWithArguments('arg', 0, value);\n```\n\n",
          },
          kindString: 'Parameter',
        },
      ],
    };
    const result = global.compileTemplate('member.parameters.hbs', context);
    expect(result).toMatchSnapshot();
  });
});
