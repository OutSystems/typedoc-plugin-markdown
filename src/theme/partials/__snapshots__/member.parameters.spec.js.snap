// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[partial] member.parameters should compile 1`] = `
"**Parameters:**

| Name | Type | Description |
| ------ | ------ | ------ |
| paramZ | [type] |  This is a string parameter. |
| paramG | [type] |  This is a parameter flagged with any.<br><br>    This sentence is placed in the next line.|
| paramA | [type] |This is a **parameter** pointing to an interface.<br><br>\`\`\`var value:BaseClass = new BaseClass('test');functionWithArguments('arg', 0, value);\`\`\`<br><br> |"
`;
